### ISSUE TITLE
"Your profile" window displays nothing, just gear wheel (2.1.23)

### ISSUE THREAD

**Issue by djvnsk** [nilsteampassnet/TeamPass#949]
![- 06 04 2015 - 16-39-55 profile](https://cloud.githubusercontent.com/assets/3201494/7004993/ffc9fb4c-dc7b-11e4-8071-71ad8957d462.png)

**Comment by N-Mi** [comment]
Same issue here, same version.

**Comment by N-Mi** [comment]
I just tried to access the Profile dialog with the Firebug console activated.

Seems there's a jquery HTTP request being blocked by Firefox :
```
Blocked loading mixed active content "http://teampass.mydomain.com/profile.php?key=FSgmSdTrkZV6Y42ACQXBmD0JomLXR2zKKVqj/kcq5Jkv68ypcV"
	

xhr.open( s.type, s.url, s.async );

jquery.js (line 8672)
```

Which lead me to this article :
https://blog.mozilla.org/tanvi/2013/04/10/mixed-content-blocking-enabled-in-firefox-23/


If I disable mixed content protection has explained on this page, I have other error messages in firebug :

```

GET http://teampass.mydomain.com/profile.php?key=FSgmSdTrkZV6Y42ACQXBmD0JomLXR2zKKVqj/kcq5Jkv68ypcV
	
301 Moved Permanently
		69ms	
jquery.js (line 8706)
Loading mixed (insecure) active content "http://teampass.mydomain.com/profile.php?key=FSgmSdTrkZV6Y42ACQXBmD0JomLXR2zKKVqj/kcq5Jkv68ypcV" on a secure page
	

xhr.open( s.type, s.url, s.async );

jquery.js (line 8672)
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://teampass.mydomain.com/profile.php?key=FSgmSdTrkZV6Y42ACQXBmD0JomLXR2zKKVqj/kcq5Jkv68ypcV. (Reason: CORS header 'Access-Control-Allow-Origin' missing).
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://teampass.mydomain.com/profile.php?key=FSgmSdTrkZV6Y42ACQXBmD0JomLXR2zKKVqj/kcq5Jkv68ypcV. (Reason: CORS request failed).

```

**Comment by nilsteampassnet** [comment]
​Can you provide the error returned by server using Firebug please?​

**Comment by N-Mi** [comment]
I'm not sure I understand what error message you ask me. Here is the content of the HTTP/301 server answer :

```
Connection		Keep-Alive
Content-Encoding	gzip
Content-Length		305
Content-Type		text/html; charset=iso-8859-1
Date			Wed, 02 Sep 2015 09:21:12 GMT
Keep-Alive		timeout=5, max=100
Location		https://mydomain.com/profile.php?key=lOPrYIt8oJhMGWaz6x9Ay/w9iLXNr0fn3y3Sg5i8uO0UI6n/rq
Server			Apache/2.2.22 (Debian)
Vary			Accept-Encoding
```

The root issue I see is the initial request is the initial jquery request is done in HTTP, while Teampass is accessed in HTTPS.

**Comment by N-Mi** [comment]
Finally, I've found how to fix this issue.

Logged as "admin". Went to "Setup" -> "TeamPass Settings".
Changed all the URL fields to HTTPS URLs instead of HTTP.
Logged out, then logged in.

On my Apache server, the vhost was reachable on both HTTP and HTTPS, but HTTP requests were permanently redirected to HTTPS.

Maybe TeamPass should guess whether to use HTTP/HTTPS for its jquery requests, depending on how the user has connected to TeamPass ?
